kinto_fxa/utils.py

```diff
@@ -1,3 +1,38 @@
+from pyramid.exceptions import ConfigurationError
+from collections import OrderedDict
+
+
 def fxa_conf(request, name):
     key = 'fxa-oauth.%s' % name
     return request.registry.settings[key]
+
+
+def parse_resources(settings):
+    resources = OrderedDict()
+    scope_routing = {}
+
+    for setting_key, setting_value in settings.items():
+        if not setting_key.startswith('fxa-oauth.'):
+            continue
+        parts = setting_key.split('.', 3)
+        if len(parts) == 2:  # default settings
+            client_name = 'default'
+            _, setting_basename = parts
+        else:
+            _, client_name, setting_basename = parts
+
+        if setting_basename == 'required_scope':
+            if setting_value in scope_routing:
+                message = '{} is already required for another config.'.format(setting_value)
+                raise ConfigurationError(message)
+            if setting_value is not None:
+                scope_routing[setting_value] = client_name
+
+        if client_name in ('relier', 'state', 'webapp'):
+            setting_basename = '{}.{}'.format(client_name, setting_basename)
+            client_name = 'default'
```

Comment:
This mixing of client names with other config parameters seems kinda fragile to me.  What if a future version of this library introduces a new config parameter, that happens to share a name with a client in my config?

How would it look if, instead, we used something like "fxa-oauth.clients.lockbox.required_scope" in order to ensure there can't be confusion between the client names and other config names?